### PR TITLE
En_Ex_Item

### PR DESCRIPTION
**Pull request by petrie911**
Also added its enums to Bom_Bowl and Changer.

### CODE REVIEW

**Review by fig02 — APPROVED** [review]
imo the ENEXITEM_ prefix can be shortened to just EX or something, but we're gonna rename the actors anyway so its whatever

and id maybe say BOWLING instead of BOWL when relevant

**Comment by petrie911** [comment]
Shortening to EXITEM is natural, though just EX conflicts with ExRuppy. Also changed the bowling thing.

File: src/overlays/actors/ovl_En_Ex_Item/z_en_ex_item.h [hunk 1]
```diff
@@ -6,13 +6,56 @@
...
+    /* 11 */ EXITEM_BLUE_RUPEE_CHEST,
+    /* 12 */ EXITEM_RED_RUPEE_CHEST,
+    /* 13 */ EXITEM_13,
+    /* 14 */ EXITEM_14,
+    /* 15 */ EXITEM_SMALL_KEY_CHEST,
+    /* 16 */ EXITEM_MAGIC_FIRE,
+    /* 17 */ EXITEM_MAGIC_WIND,
+    /* 18 */ EXITEM_MAGIC_DARK,
+    /* 19 */ EXITEM_BULLET_BAG
+} EnExItemParam;
```
Comment (Roman971) [inline src/overlays/actors/ovl_En_Ex_Item/z_en_ex_item.h:55]:
```suggestion
} EnExItemType;
```

File: src/overlays/actors/ovl_En_Ex_Item/z_en_ex_item.c [hunk 2]
```diff
@@ -21,39 +44,477 @@ const ActorInit En_Ex_Item_InitVars = {
...
+    f32 tmpf3;
+    f32 tmpf4;
+    f32 tmpf5;
+    f32 tmpf6;
+    f32 tmpf7;
+    f32 sp3C;
+
+    if (!this->stopRotate) {
+        this->actor.shape.rot.y += 0x1000;
+        if ((this->prizeRotateTimer == 0) && ((this->actor.shape.rot.y & 0xFFFF) == 0x9000)) {
```
Comment (Roman971) [inline src/overlays/actors/ovl_En_Ex_Item/z_en_ex_item.c:271]: Is the mask really necessary here? (I'd imagine it comes from the += right before)
```suggestion
        if ((this->prizeRotateTimer == 0) && (this->actor.shape.rot.y == 0x9000)) {
```

File: src/overlays/actors/ovl_En_Ex_Item/z_en_ex_item.c [hunk 3]
```diff
@@ -21,39 +44,477 @@ const ActorInit En_Ex_Item_InitVars = {
...
+    f32 tmpf3;
+    f32 tmpf4;
+    f32 tmpf5;
+    f32 tmpf6;
+    f32 tmpf7;
+
+    Math_ApproachF(&this->scale, 0.8f, 0.1f, 0.02f);
+    if (!this->stopRotate) {
+        this->actor.shape.rot.y += 0x1000;
+        if ((this->prizeRotateTimer == 0) && ((this->actor.shape.rot.y & 0xFFFF) == 0x9000)) {
```
Comment (Roman971) [inline src/overlays/actors/ovl_En_Ex_Item/z_en_ex_item.c:357]: Same suggestion here
```suggestion
        if ((this->prizeRotateTimer == 0) && (this->actor.shape.rot.y == 0x9000)) {
```

File: src/overlays/actors/ovl_En_Ex_Item/z_en_ex_item.c [hunk 4]
```diff
@@ -21,39 +44,477 @@ const ActorInit En_Ex_Item_InitVars = {
...
+    static s16 drawItemIds[] = { 0x63, 0x64, 0x65 };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ex_Item/EnExItem_Draw.s")
+    func_8002ED80(&this->actor, globalCtx, 0);
+    func_800694A0(globalCtx, drawItemIds[magicIndex]);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Ex_Item/func_80A09B6C.s")
+void EnExItem_DrawKey(EnExItem* this, GlobalContext* globalCtx, s32 index) {
+    static s32 keySegment[] = { 0x0403F140 };
```
Comment (Roman971) [inline src/overlays/actors/ovl_En_Ex_Item/z_en_ex_item.c:499]:
```suggestion
    static s32 keySegments[] = { 0x0403F140 };
```

**Review by Roman971 — COMMENTED** [review]
Just a few minor things

File: src/overlays/actors/ovl_En_Ex_Item/z_en_ex_item.c (hunk 2 above)
Comment (petrie911) [reply src/overlays/actors/ovl_En_Ex_Item/z_en_ex_item.c:271]: Can't get it to work without that. Note that shape.rot.y is a `s16`, so it wouldn't cast with `&0xFFFF`. That said, even using a `(u16)` cast doesn't work. It seems to have to be the `&0xFFFF`.

Comment (Roman971) [reply src/overlays/actors/ovl_En_Ex_Item/z_en_ex_item.c:271]: Right, I forgot these were `s16`s, nvm

**Review by Roman971 — APPROVED** [review]

**Pull request merged**